presentation/components/src/test/instance-filter-builder/PresentationInstanceFilterDialog.test.tsx

```diff
@@ -0,0 +1,176 @@
+    Element.prototype.scrollIntoView = sinon.stub();
+  });
+
+  after(async () => {
+    Presentation.terminate();
+    UiComponents.terminate();
+    await IModelApp.shutdown();
+    sinon.restore();
+  });
+
+  it("invokes 'onInstanceFilterApplied' with filter", async () => {
+    const spy = sinon.spy();
+    const { container, getByText, getByDisplayValue } = render(<PresentationInstanceFilterDialog
+      imodel={imodelMock.object}
+      descriptor={descriptor}
+      onClose={() => { }}
+      onApply={spy}
+      isOpen={true} />);
+
+    const applyButton = container.querySelector<HTMLInputElement>(".presentation-instance-filter-button-bar .iui-high-visibility:disabled");
+    expect(applyButton?.disabled).to.be.true;
+
+    // select property
+    const propertySelector = container.querySelector<HTMLInputElement>(".rule-property .iui-input");
+    expect(propertySelector).to.not.be.null;
+    propertySelector?.focus();
+    fireEvent.click(getByText(propertiesField.label));
+    // wait until property is selected
+    await waitFor(() => getByDisplayValue(propertiesField.label));
+    // select operator
+    const operatorSelector = container.querySelector<HTMLInputElement>(".rule-operator .iui-select-button");
+    expect(operatorSelector).to.not.be.null;
+    fireEvent.click(operatorSelector!);
+    fireEvent.click(getByText(getPropertyFilterOperatorLabel(PropertyFilterRuleOperator.IsNotNull)));
+    // wait until operator is selected
+    await waitFor(() => getByText(getPropertyFilterOperatorLabel(PropertyFilterRuleOperator.IsNotNull)));
+
+    expect(applyButton?.disabled).to.be.false;
+    fireEvent.click(applyButton!);
+    expect(spy).to.be.calledOnceWith({
+      filter: {
+        field: propertiesField,
+        operator: PropertyFilterRuleOperator.IsNotNull,
+        value: undefined,
+      },
+      usedClasses: [classInfo],
+    });
+  });
+
+  it("invokes 'onInstanceFilterApplied' with initial filter", async () => {
+    const spy = sinon.spy();
+    const { container } = render(<PresentationInstanceFilterDialog
+      imodel={imodelMock.object}
+      descriptor={descriptor}
+      onClose={() => { }}
+      onApply={spy}
+      isOpen={true}
+      initialFilter={initialFilter} />);
+
+    const applyButton = container.querySelector<HTMLInputElement>(".presentation-instance-filter-button-bar .iui-high-visibility");
+    expect(applyButton?.disabled).to.be.false;
+
+    fireEvent.click(applyButton!);
+    expect(spy).to.be.calledOnceWith(initialFilter);
+  });
+
+  it("renders custom title", async () => {
+    const spy = sinon.spy();
+    const title = "custom title";
+
+    const { queryByText } = render(<PresentationInstanceFilterDialog
+      imodel={imodelMock.object}
+      descriptor={descriptor}
+      onClose={() => { }}
+      title={<div>{title}</div>}
+      onApply={spy}
+      isOpen={true}
+      initialFilter={initialFilter} />);
+
+    expect(queryByText(title)).to.not.be.null;
+
```

Comment:
Looks like there are some unnecessary empty lines left in this file. Please remove them